src/cuga/backend/cuga_graph/nodes/cuga_agent_core/graph/shared_nodes.py

```
@@ -0,0 +1,249 @@
+        adapter.on_response_processed(state, code, content)
+
+        # ── Tool-approval interrupt for generated code ─────────────────────
+        if code and settings.policy.enabled:
+            approval_command = await ToolApprovalHandler.check_and_create_approval_interrupt(
+                adapter, state, code, content, config
+            )
+            if approval_command:
+                return approval_command
+
+        # ── Build final message list + step count ──────────────────────────
+        final_messages: list = modified_messages + [AIMessage(content=content)]
```

Ruff (0.15.13)

[warning] Consider `[*modified_messages, AIMessage(content=content)]` instead of concatenation

Replace with `[*modified_messages, AIMessage(content=content)]`

(RUF005)